framework/src/vectorpostprocessors/ElementVariablesDifferenceMax.C

```diff
@@ -0,0 +1,123 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ElementVariablesDifferenceMax.h"
+
+// MOOSE includes
+#include "MooseVariable.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+template <>
+InputParameters
+validParams<ElementVariablesDifferenceMax>()
+{
+  InputParameters params = validParams<ElementVectorPostprocessor>();
+
+  params.addRequiredCoupledVar(
+      "compare_a",
+      "The first variable to evaluate the difference with, performed as \"compare_a - compare_b\"");
+  params.addRequiredCoupledVar("compare_b",
+                               "The second variable to evaluate the difference with, "
+                               "performed as \"compare_a - compare_b\"");
+
+  params.addParam<bool>(
+      "furthest_from_zero", false, "Find the difference with the highest absolute value");
+
+  return params;
+}
+
+ElementVariablesDifferenceMax::ElementVariablesDifferenceMax(const InputParameters & parameters)
+  : ElementVectorPostprocessor(parameters),
+    _a(coupledValue("compare_a")),
+    _b(coupledValue("compare_b")),
+    _a_value(declareVector(getVar("compare_a", 0)->name())),
+    _b_value(declareVector(getVar("compare_b", 0)->name())),
+    _max_difference(declareVector("Difference")),
+    _position_x(declareVector("X")),
+    _position_y(declareVector("Y")),
+    _position_z(declareVector("Z")),
+    _furthest_from_zero(getParam<bool>("furthest_from_zero"))
+{
+  // These are all single-value arrays
+  _a_value.resize(1);
+  _b_value.resize(1);
+  _max_difference.resize(1);
+  _position_x.resize(1);
+  _position_y.resize(1);
+  _position_z.resize(1);
+}
+
+void
+ElementVariablesDifferenceMax::execute()
+{
+  for (unsigned int qp = 0; qp < _qrule->n_points(); ++qp)
+  {
+    // Get the difference
+    const Real difference = _furthest_from_zero ? std::abs(_a[qp] - _b[qp]) : _a[qp] - _b[qp];
```

Comment:
No it's not although that symmetry problem is actually a bigger issue as it's more than just symmetry that can get you. Any time you have multiple values at the extreme you are going to run into this problem so you actually need to decide on how you want to solve it. One suggestion would be to keep track of yet another piece of data such as the element or node ID and keep the one that's highest or lowest or something. Of course that also means you'll have to communicate that data in parallel since it could bounce around if all the processors don't agree on the global location of the min/max.

I'll try to illustrate again by marking up more of this code...